src/VisualStudio/IntegrationTest/IntegrationTests/CSharp/CSharpFindReferences.cs

```diff
@@ -0,0 +1,122 @@
+using Microsoft.VisualStudio.IntegrationTest.Utilities;
+using Microsoft.VisualStudio.IntegrationTest.Utilities.Input;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Roslyn.VisualStudio.IntegrationTests.CSharp
+{
+    [Collection(nameof(SharedIntegrationHostFixture))]
+    public class CSharpFindReferences : AbstractEditorTest
+    {
+        protected override string LanguageName => LanguageNames.CSharp;
+
+        public CSharpFindReferences(VisualStudioInstanceFactory instanceFactory)
+            : base(instanceFactory, nameof(CSharpFindReferences))
+        {
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.FindReferences)]
+        public void FindReferencesToCtor()
+        {
+            SetUpEditor(@"
+class Program
+{
+}$$
+");
+
+            VisualStudio.Instance.SolutionExplorer.AddFile(ProjectName, "File2.cs");
+            VisualStudio.Instance.SolutionExplorer.OpenFile(ProjectName, "File2.cs");
+
+            SetUpEditor(@"
+class SomeOtherClass
+{
+    void M()
+    {
+        Program p = new Progr$$am();
+    }
+}
+");
+
+            SendKeys(Shift(VirtualKey.F12));
+
+            VisualStudioWorkspaceOutOfProc.WaitForAsyncOperations(FeatureAttribute.FindReferences);
+
+            const string programReferencesCaption = "'Program' references";
+            var activeWindowCaption = VisualStudio.Instance.Shell.GetActiveWindowCaption();
+            Assert.Equal(expected: programReferencesCaption, actual: activeWindowCaption);
+
+            var results = VisualStudio.Instance.FindReferencesWindow.GetContents(programReferencesCaption);
+
+            var reference = results.Single();
+            Assert.Equal(expected: "Program p = new Program();", actual: reference.Code);
+            Assert.Equal(expected: 5, actual: reference.Line);
+            Assert.Equal(expected: 24, actual: reference.Column);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.FindReferences)]
+        public void FindReferencesToLocals()
+        {
+            SetUpEditor(@"
+class Program
+{
+    static void Main()
+    {
+        int local = 1;
+        Console.WriteLine(local$$);
+    }
+}
+");
+
+            SendKeys(Shift(VirtualKey.F12));
+
+            VisualStudioWorkspaceOutOfProc.WaitForAsyncOperations(FeatureAttribute.FindReferences);
+
+            const string localReferencesCaption = "'local' references";
+            var activeWindowCaption = VisualStudio.Instance.Shell.GetActiveWindowCaption();
+            Assert.Equal(expected: localReferencesCaption, actual: activeWindowCaption);
+
+            var results = VisualStudio.Instance.FindReferencesWindow.GetContents(localReferencesCaption);
+
+            var reference = results.Single();
+            Assert.Equal(expected: "Console.WriteLine(local);", actual: reference.Code);
+            Assert.Equal(expected: 6, actual: reference.Line);
+            Assert.Equal(expected: 26, actual: reference.Column);
```

Comment:
thesee tests have lots of repeated code.  can we extract this out to a general FindRefsTestHelper that all the tests would call?